### ISSUE TITLE
It is not work when using file() to read files in the NFS in clickhouse-server-23.11.2.11

### ISSUE THREAD

**Issue by babu1801** [ClickHouse/ClickHouse#57965]
**Describe the unexpected behaviour**

when selecting data through file() function,  the query failed, and get the exception:
**Code: 291   File  `...` is not inside  '/home/clickhouse/ckhome/user_files/'  (DATABASE_ACCESS_DENIED)**   

the files  are saved in a NFS  path, which is mounted on the clickhouse host.
I create a soft link under the  path defined in <user_files_path>. 

**Before 23.11.1 and 23.11.2, this method is work**. I can load data in the files on the NFS through file().
Begin from 23.11.1，it is not worked any more.

**How to reproduce**
mount a NFS filesystem in the Clickhouse server,  for example : /data/RawTradeData

and create a soft link under the user file path defined in the  <user_files_path>/home/clickhouse/ckhome/user_files/</user_fiels_path>

        ln -s /data/RawTradeData /home/clickhouse/ckhome/user_files/RawTradeData

after that,  query data:
select * from file('RawTradeData/2023/ServerHX14/**/PubOrderLog_20231121.csv', 'CSVWithNames');

it will get results before 23.11.1 , and get error message since 23.11.1

* Which ClickHouse server version to use
    clickhouse-server-23.11.2.11 

* Which interface to use, if matters
 clickhouse-client , or  DataGrip through JDBC

* Queries to run that lead to unexpected result
select * from file('RawTradeData/2023/ServerHX14/**/PubOrderLog_20231121.csv', 'CSVWithNames');

**Expected behavior**
the data in the csv will be returned

**Error message and/or stacktrace**
![image](https://github.com/ClickHouse/ClickHouse/assets/19375576/38aa2aec-0fa4-47db-bfc2-8e2ca50a494b)

![image](https://github.com/ClickHouse/ClickHouse/assets/19375576/a85ae834-d053-46db-9c87-c02ded4c0bf0)

![c2560f2f55e9cbde6a56d810a446f77](https://github.com/ClickHouse/ClickHouse/assets/19375576/d81dbc51-4639-4720-b61e-3f6dde54559a)

**Additional context**

**Comment by alexey-milovidov** [comment]
@zvonand is it related to a different implementation of globs?

**Comment by zvonand** [comment]
> @zvonand is it related to a different implementation of globs?

Not sure. I'll take a look

**Comment by zvonand** [comment]
I don't think it is due to my changes -- it works fine with usual symlinks on local FS.  
I'll try to reproduce it with NFS
